src/tqec/compile/block.py

```diff
@@ -34,26 +34,28 @@ class CompiledBlock:
     def num_layers(self) -> int:
         return len(self.layers)
 
-    def with_updated_layer(
+    def update_layers(
         self,
-        plaquettes_to_update: Mapping[int, Plaquette],
-        layer_to_update: int,
-    ) -> CompiledBlock:
-        """Returns a new `CompiledBlock` with the specified layer updated.
+        substitution: Mapping[int, Plaquettes],
+    ) -> None:
+        """Update the plaquettes in a specific layer of the block.
```

Comment:
Minor comment: it might be worth explicitly writing down that the modification is performed in-place.